src/main/java/frc/robot/superstructure/Superstructure.java

```diff
@@ -127,71 +178,125 @@ public Command speaker() {
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
+			new SequentialCommandGroup(
+				elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR).until(this::isNoteInElevatorRoller),
+				elevatorRollerStateHandler.setState(ElevatorRollerState.STOP)
+			),
 			flywheelStateHandler.setState(FlywheelState.DEFAULT),
-			elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR),
-			funnelStateHandler.setState(FunnelState.SHOOTER_TO_ELEVATOR),
-			intakeStatesHandler.setState(IntakeStates.SHOOTER_TO_ELEVATOR),
-			pivotStateHandler.setState(PivotState.UP),
-			elevatorStatesHandler.setState(ElevatorStates.IDLE)
+			elevatorStatesHandler.setState(ElevatorStates.IDLE),
+			pivotStateHandler.setState(PivotState.UP)
 		);
 	}
 
 	public Command transferElevatorShooter() {
 		return new ParallelCommandGroup(
+			setCurrentStateValue(RobotState.TRANSFER_ELEVATOR_SHOOTER),
+			new SequentialCommandGroup(
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP)
+				),//.until(isReadyToTransfer)
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.ELEVATOR_TO_SHOOTER),
+					funnelStateHandler.setState(FunnelState.NOTE_TO_SHOOTER)
+				).until(this::isNoteInShooter),
+				new ParallelCommandGroup(
+					intakeStatesHandler.setState(IntakeStates.STOP),
+					funnelStateHandler.setState(FunnelState.STOP)
+				)
+			),
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
+			new SequentialCommandGroup(
+				elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_TO_ELEVATOR).until(() -> !this.isNoteInElevatorRoller()),
+				elevatorRollerStateHandler.setState(ElevatorRollerState.STOP)
+			),
 			flywheelStateHandler.setState(FlywheelState.DEFAULT),
-			elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_FROM_ELEVATOR),
-			funnelStateHandler.setState(FunnelState.NOTE_TO_SHOOTER),
-			intakeStatesHandler.setState(IntakeStates.NOTE_TO_SHOOTER),
-			pivotStateHandler.setState(PivotState.UP),
-			elevatorStatesHandler.setState(ElevatorStates.IDLE)
+			elevatorStatesHandler.setState(ElevatorStates.IDLE),
+			pivotStateHandler.setState(PivotState.UP)
 		);
 	}
 
 	public Command intakeOuttake() {
 		return new ParallelCommandGroup(
+			setCurrentStateValue(RobotState.INTAKE_OUTTAKE),
 			swerve.getCommandsBuilder().saveState(SwerveState.DEFAULT_DRIVE),
-			flywheelStateHandler.setState(FlywheelState.DEFAULT),
 			elevatorRollerStateHandler.setState(ElevatorRollerState.TRANSFER_FROM_ELEVATOR),
+			flywheelStateHandler.setState(FlywheelState.DEFAULT),
 			funnelStateHandler.setState(FunnelState.INTAKE_OUTTAKE),
 			intakeStatesHandler.setState(IntakeStates.OUTTAKE),
 			pivotStateHandler.setState(PivotState.ON_FLOOR),
 			elevatorStatesHandler.setState(ElevatorStates.IDLE)
-		);
+		).until(() -> !isNoteInIntake() && !isNoteInElevatorRoller() && !isNoteInShooter());
```

Comment:
Maybe create a isNoteIn function

Reply:
i dont like in outtake the condition, because the note can be in robot but not be seen by any sensor

Reply:
Yeah i agree, maybe we would assume the note is already in the shooter/intake (בהתאם to The func) and just check if it isn't there?